Document/0x05b-Basic-Security_Testing.md

```diff
@@ -774,24 +773,259 @@ Note that if you have the original source code and use Android Studio, you do no
+
+- Which files does the app include inside its package?
+- Which native libraries does the app use?
+- Which app components does the app use? Any services or content providers?
+- Is the app debuggable?
+- Does the app create any new files when being installed?
+
+##### Listing Installed Apps
+
+When targeting apps that are installed on the device first you'll have to decide which app you'd like to analyze. You can retrieve the installed apps either by using `pm` (Android Package Manager) or by using `frida-ps`:
+
+```bash
+$ adb shell pm list packages
+package:sg.vantagepoint.helloworldjni
+package:eu.chainfire.supersu
+package:org.teamsik.apps.hackingchallenge.easy
+package:org.teamsik.apps.hackingchallenge.hard
+package:sg.vp.owasp_mobile.omtg_android
+```
+
+You can include flags to show only third party apps (`-3`) and the location of their APK file (`-f`), which you can use afterwards to download it via `adb pull`:
+
+```bash
+$ adb shell pm list packages -3 -f
+package:/data/app/sg.vantagepoint.helloworldjni-1/base.apk=sg.vantagepoint.helloworldjni
+package:/data/app/eu.chainfire.supersu-1/base.apk=eu.chainfire.supersu
+package:/data/app/org.teamsik.apps.hackingchallenge.easy-1/base.apk=org.teamsik.apps.hackingchallenge.easy
+package:/data/app/org.teamsik.apps.hackingchallenge.hard-1/base.apk=org.teamsik.apps.hackingchallenge.hard
+package:/data/app/sg.vp.owasp_mobile.omtg_android-kR0ovWl9eoU_yh0jPJ9caQ==/base.apk=sg.vp.owasp_mobile.omtg_android
+```
+
+This is the same as running `adb shell pm path <app_package_id>` on an app package ID:
+
+```bash
+$ adb shell pm path sg.vp.owasp_mobile.omtg_android
+package:/data/app/sg.vp.owasp_mobile.omtg_android-kR0ovWl9eoU_yh0jPJ9caQ==/base.apk
+```
+
+Use `frida-ps -Uai` to get all apps (`-a`) currently installed (`-i`) on the connected USB device (`-U`):
+
+```bash
+$ frida-ps -Uai
+  PID  Name                                      Identifier
+-----  ----------------------------------------  ---------------------------------------
+  766  Android System                            android
+21228  Attack me if u can                        sg.vp.owasp_mobile.omtg_android
+ 4281  Termux                                    com.termux
+    -  Uncrackable1                              sg.vantagepoint.uncrackable1
+    -  drozer Agent                              com.mwr.dz
+```
+
+Note that this also shows the PID of the apps that are running at the moment. Take a note of the "Identifier" and the PID if any as you'll need them afterwards.
+
+##### Exploring the App Package
+
+Once you are targeting an specific app you'll want to start gathering information about it. At this point you might have the APK of the app.
+As you've seen above, installed Android apps are located at `/data/app/[package-name]` and their APKs can be retrieved from there.
```

Comment:
And remove this line, as it will be explained in the referenced section.

Reply:
done